### PR TITLE
chore: add GH PR check to validate list of vsix files definition

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: Valerii Svydenko <vsvydenk@redhat.com>

### What does this PR do?
Add GH PR check to validate the definition of VS Code extensions

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-3477

### CODE REVIEW

File: dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    # version should be set if download link was set
+    if [[ $vsixDownloadLink != null ]] && [[ $vsixVersion == null ]]; then
+      echo -e "Version is not defined for $vsixFullName"
+      echo -e "${RED}${EMOJI_FAIL}${RESETSTYLE} Test failed!"
+      exit 1
+    fi
+    
+    # Publisher and name should be divided by . in extension's id
+    if [[ $vsixFullName != *.* ]]; then
+      echo -e "Publisher and name should be divided by . for $vsixFullName"
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh:66]: divided by `.` ?

your check is just for "string has a ." but doesn't verify that it's used as a separator between two substrings... does that matter?

File: dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+#
+# Copyright (c) 2018-2022 Red Hat, Inc.
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh:3]: copyright should be 2022

File: dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh:9]: could this script explain what it does?

File: .github/workflows/vsix-definition-pr-check.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2022 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+name: VSIX Definition PR Check
```
Comment (nickboldt) [inline .github/workflows/vsix-definition-pr-check.yaml:10]: do you want this new check linked as a badge from the README (mostly thinking about upstream... not sure if we have README action badges in downstream).

**Review by nickboldt — APPROVED** [review]

File: dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh (hunk 1 above)
Comment (svor) [reply dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh:69]: yes, it does make sense, fixed!

File: dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh (hunk 2 above)
Comment (svor) [reply dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh]: fixed

File: dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh (hunk 3 above)
Comment (svor) [reply dependencies/che-plugin-registry/build/scripts/test/validate_vsix_list.sh]: added

File: .github/workflows/vsix-definition-pr-check.yaml (hunk 4 above)
Comment (svor) [reply .github/workflows/vsix-definition-pr-check.yaml:10]: it's just a PR check, I don't think it will be useful in Readme file

**Pull request merged**